### PR TITLE
Fix a few IndexOf{Any} uses

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]
Does this suggest missing API - ContainsAny on Span, and maybe IndexOfAny(char, char..)

File: src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs [hunk 1]
```diff
@@ -96,10 +96,7 @@ private static void AddPrefixInternal(string p, HttpListener listener)
             if (lp.Host != "*" && lp.Host != "+" && Uri.CheckHostName(lp.Host) == UriHostNameType.Unknown)
                 throw new HttpListenerException((int)HttpStatusCode.BadRequest, SR.net_listener_host);
 
-            if (lp.Path!.Contains('%'))
-                throw new HttpListenerException((int)HttpStatusCode.BadRequest, SR.net_invalid_path);
-
-            if (lp.Path.IndexOf("//", StringComparison.Ordinal) != -1)
```
Comment (yyjdelete) [inline src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs:102]: Is it right here? "//" and '/' looks like different.

File: src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs [hunk 2]
```diff
@@ -203,10 +200,7 @@ public static void RemovePrefix(string prefix, HttpListener listener)
         private static void RemovePrefixInternal(string prefix, HttpListener listener)
         {
             ListenerPrefix lp = new ListenerPrefix(prefix);
-            if (lp.Path!.Contains('%'))
-                return;
-
-            if (lp.Path.IndexOf("//", StringComparison.Ordinal) != -1)
```
Comment (yyjdelete) [inline src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs:209]: The same.

File: src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs:102]: Nope, this is a mistake.

**Comment by stephentoub** [comment]
I'll fix the registry case in a different pr when I'm awake :-)

**Comment by danmoseley** [comment]
Apparently I'm not awake either

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @stephentoub !

**Pull request merged**